client/src/pages/Main.tsx

```diff
@@ -12,14 +13,14 @@ interface Message {
     content: string;
 }
 
-const Main: React.FC<MainProps> = ({ messageCount, messagePosition, messageDisappearTime }) => {
+const Main: React.FC<MainProps> = ({ eventSource, messageCount, messagePosition, messageDisappearTime }) => {
```

Comment:
The Main component no longer initialised EventSource connection. It retrieves it from the root component.